mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -160,6 +211,16 @@ pub enum EndpointError {
     InvalidUri(#[from] InvalidUriError),
     #[error("endpoint could not be resolved")]
     UnresolvedEndpoint(#[from] ResolverError),
+    #[error("Endpoint properties not resolved")]
+    SerdeError(serde_json::Error),
```

Comment:
nit: `Serde` shouldn't be in the name, and we want the `from` part too
```suggestion
    ParseError(#{from] serde_json::Error),
```